npm/asciidoctor-revealjs

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env node
+
+'use strict'
+
+process.title = 'asciidoctor-revealjs'
+
+const { Options, Invoker } = require('@asciidoctor/cli')
+const asciidoctorRevealjs = require('../dist/main.js')
+asciidoctorRevealjs.register()
+
+const options = new Options({backend: 'revealjs'}).parse(process.argv)
+
+class RevealJsInvoker extends Invoker {
+  version () {
+    const pkg = require('../package.json')
+    return `Asciidoctor-Reveal.js ${pkg.version} using ${super.version()}`
```

Comment:
I think the project name should be "Asciidoctor reveal.js" no?
```suggestion
    return `Asciidoctor reveal.js ${pkg.version} using ${super.version()}`
```

Reply:
I thought a lot about this before changing it from your code proposal. Here's the rundown:

* A project name rarely has spaces in them
* the space could make it non-obvious that it's the same thing, confusing users
* Most slide post-processors in the Asciidoctor ecosystem use a project name with dashes (bespoke)
* Reveal.js should be capitalized (I was wrong)

Then, re-evaluating:

* upstream reveal.js is never capitalized, it is actually proudly not-so, so this is accepted
* it is true that it is a converter on-top of asciidoctor so the space might represent that
* yet the command-line program is called with a dash...
* but without the dot...

`Asciidoctor-reveal.js` or `Asciidoctor reveal.js` what a harmless decision yet so difficult... I'll sleep on it.

Reply:
A project has a name and an identifier. From what I've seen the rules are:

**Project name**
- Respect the upstream project capitalization
- Write Asciidoctor with a capital A
- Use a space between Asciidoctor and the upstream project

**Project identifier**
- If possible the ".js" extension should be dropped.
- The project identifier should be lower-kebab-case (ie. `asciidoctor-bespoke`, `asciidoctor-pdf`, `asciidoctor-epub3`).
- The project identifier should be used to name:
   * the binary
   * the project on GitHub
   * the Gem
   * the npm package

Here's an example:

**Asciidoctor Bespoke**
- Upstream project: Bespoke.js
- Project name: Asciidoctor Bespoke
- Project identifier: `asciidoctor-bespoke`
   * Gem: `asciidoctor-bespoke`
   * GitHub project: https://github.com/asciidoctor/asciidoctor-bespoke
   * Binary: `asciidoctor-bespoke`

References:
- [bin/asciidoctor-bespoke](https://github.com/asciidoctor/asciidoctor-bespoke/blob/d402d631c432bc641cce8d4a4f477c18ccafd758/bin/asciidoctor-bespoke#L14)
- [asciidoctor-bespoke.gemspec](https://github.com/asciidoctor/asciidoctor-bespoke/blob/d402d631c432bc641cce8d4a4f477c18ccafd758/asciidoctor-bespoke.gemspec#L5)
- [Bespoke.js](http://markdalgleish.com/projects/bespoke.js/)

Please note that the above rules also apply to:
- https://github.com/asciidoctor/asciidoctor-epub3
- https://github.com/asciidoctor/asciidoctor-pdf

Reply:
> Asciidoctor-reveal.js or Asciidoctor reveal.js what a harmless decision yet so difficult... I'll sleep on it.

It should be: "Asciidoctor reveal.js using ". See:

- [bin/asciidoctor-bespoke](https://github.com/asciidoctor/asciidoctor-bespoke/blob/d402d631c432bc641cce8d4a4f477c18ccafd758/bin/asciidoctor-bespoke#L14)
- [bin/asciidoctor-epub3](https://github.com/asciidoctor/asciidoctor-epub3/blob/27b8f124020bf143e2ee7ce916f220a06fbadeb4/bin/asciidoctor-epub3#L15)
- [bin/asciidoctor-pdf](https://github.com/asciidoctor/asciidoctor-pdf/blob/f71116b0060ef896e84dd94f6d4126e6b87fea98/bin/asciidoctor-pdf#L15)

Reply:
As a side note, we should probably update the Ruby binary [bin/asciidoctor-revealjs](https://github.com/asciidoctor/asciidoctor-reveal.js/blob/master/bin/asciidoctor-revealjs) to add the project name and version.

Reply:
> what a harmless decision yet so difficult

Sorry I might have a small obsession on capitalization :laughing:

Reply:
You've convinced me.